todo_list/base/templates/base/task_list.html

```diff
@@ -0,0 +1,48 @@
+{% extends 'base/main.html' %}
+{% block content %}
+{% load static %}
+
+<div class="header-bar">
+    <div>
+        <h1>Hello, {{request.user|title}}</h1>
+        <h3 style="margin:0">You have <i>{{count}} task{{count|pluralize:"s"}} to do</i></h3>
+    </div>
+    {% if request.user.is_authenticated %}
+    <a href="{% url 'logout' %}">Logout</a>
+    {% else %}
+    <a href="{% url 'login' %}">Login</a>
+    {% endif %}
+</div>
+
+<div id="search-add-wrapper">
+    <form method="GET" style="margin-top: 20px; display: flex;">
+    <input style="margin-right: 10px" type="text" name="search-area" value="{{search_input}}">
```

Comment:
```suggestion html
    <input style="margin-right: 10px" type="search" name="search-area" value="{{search_input}}">
```
Для поиска использовать тип поиск
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search